### PR TITLE
Rename Admin to AbstractAdmin

### PR DESCRIPTION
**Pull request by Soullivaneuh**
This follows Symfony coding standard rules.

Ref: http://symfony.com/doc/current/contributing/code/standards.html#naming-conventions

@sonata-project/contributors I need your review here!

Todo: Manual check on projects.

### CODE REVIEW

File: Resources/doc/cookbook/recipe_image_previews.rst [hunk 1]
```diff
@@ -52,18 +52,18 @@ we are manipulating form fields we do this from within ``ImageAdmin::configureFo
...
                 // get the container so the full path to the image can be set
                 $container = $this->getConfigurationPool()->getContainer();
                 $fullPath = $container->get('request')->getBasePath().'/'.$webPath;
-            
+
                 // add a 'help' option containing the preview's img tag
                 $fileFieldOptions['help'] = '<img src="'.$fullPath.'" class="admin-preview" />';
             }
-            
+
```
Comment (greg0ire) [inline Resources/doc/cookbook/recipe_image_previews.rst]: All these whitespace changes should be in a separate commit please (use `git add --patch`)

**Comment by greg0ire** [comment]
Not a fan.

- This will be a big hassle.
- The goal is to follow the coding standards of another project.

File: Resources/doc/cookbook/recipe_image_previews.rst (hunk 1 above)
Comment (Soullivaneuh) [inline Resources/doc/cookbook/recipe_image_previews.rst]: All those whitespace changes are auto removed by my IDE.

Comment (greg0ire) [inline Resources/doc/cookbook/recipe_image_previews.rst]: Does that prevent you from using `git add --patch` ? :trollface:

**Comment by Soullivaneuh** [comment]
> This will be a big hassle.

This is just a class rename with a deprecated warning to the upgrade will be smooth.

> The goal is to follow the coding standards of another project.

This is not "the coding standards of another project" but a coding standard adopted for many Symfony bundles.

AFAIK, sonata-project does follow Symfony standards. I don't see why we should not follow this one.

**Comment by greg0ire** [comment]
> This is just a class rename with a deprecated warning to the upgrade will be smooth.

Could you try this on a project of yours and provide a `sed` command with this PR ?

> AFAIK, sonata-project does follow Symfony standards

I didn't even notice this is enforced by StyleCI, but you're indeed right, so…

File: Resources/doc/cookbook/recipe_image_previews.rst (hunk 1 above)
Comment (Soullivaneuh) [inline Resources/doc/cookbook/recipe_image_previews.rst]: Done, commit nazi. :trollface:

File: Tests/Command/ValidatorsTest.php [hunk 2]
```diff
@@ -102,8 +102,8 @@ public function testValidateClass($expected, $value)
     public function getValidateClassTests()
     {
         return array(
-            array('Sonata\AdminBundle\Admin\Admin', 'Sonata\AdminBundle\Admin\Admin'),
-            array('Sonata\AdminBundle\Admin\Admin', 'Sonata/AdminBundle/Admin/Admin'),
+            array('Sonata\AdminBundle\Admin\AbstractAdmin', 'Sonata\AdminBundle\Admin\AbstractAdmin'),
+            array('Sonata\AdminBundle\Admin\AbstractAdmin', 'Sonata/AdminBundle/Admin/Admin'),
```
Comment (OskarStark) [inline Tests/Command/ValidatorsTest.php]: looks like this is missing,, could you please check Class names with `/`, too ?

File: Tests/Show/ShowMapperTest.php [hunk 3]
```diff
@@ -11,7 +11,6 @@
 
 namespace Sonata\AdminBundle\Tests\Show;
 
-use Sonata\AdminBundle\Admin\Admin;
```
Comment (OskarStark) [inline Tests/Show/ShowMapperTest.php]: was it unused before?

**Comment by OskarStark** [comment]
No `@author` tag from you in AbstractAdmin @Soullivaneuh ?

File: Tests/Show/ShowMapperTest.php (hunk 3 above)
Comment (Soullivaneuh) [inline Tests/Show/ShowMapperTest.php]: Looks like, yes.

**Comment by Soullivaneuh** [comment]
> No @author tag from you in AbstractAdmin @Soullivaneuh ?

Well, I just did a rename. :-)

**Comment by OskarStark** [comment]
> Well, I just did a rename. :-)

yeah, you are probably right

File: Tests/Show/ShowMapperTest.php (hunk 3 above)
Comment (OskarStark) [inline Tests/Show/ShowMapperTest.php]: 👍

**Comment by OskarStark** [comment]
LGTM @Soullivaneuh

**Comment by Soullivaneuh** [comment]
I want to test it manually on my project first.

File: Admin/AbstractAdmin.php [hunk 4]
```diff
@@ -48,11 +48,9 @@
 use Symfony\Component\Validator\ValidatorInterface as LegacyValidatorInterface;
 
 /**
- * Class Admin.
- *
```
Comment (greg0ire) [inline Admin/AbstractAdmin.php]: I knew it! You changed something else!

Comment (Soullivaneuh) [inline Admin/AbstractAdmin.php]: YOU MAD?! :trollface:
